sdk/tests/core/test_az_json.c

```diff
@@ -3010,11 +3010,25 @@ static void test_az_json_string_unescape(void** state)
     destination_span._internal.size = final_size;
 
     assert_int_equal(AZ_OK, result);
-    assert_int_equal(expected._internal.size, final_size);
-    assert_string_equal(expected._internal.ptr, destination);
     assert_true(az_span_is_content_equal(expected, destination_span));
   }
 
+  // unexpected end
+  {
+    uint8_t buffer[3];
+    az_span test_span = AZ_SPAN_FROM_BUFFER(buffer);
+    test_span._internal.ptr[0] = 'a';
+    test_span._internal.ptr[1] = '"';
```

Comment:
Generally each unit test should validate one thing.
This test case is about hitting an unexpected end. We should remove embedded quotes, as that's an invalid JSON string.

```suggestion
    test_span._internal.ptr[1] = 'b';
```

Let's separate the two test cases.

Otherwise, when we add unescaped embedded quote validation, this test will fail with unexpected char, and not unexpected end.

Reply:
done